core/src/main/scala/chisel3/StrongEnum.scala

```diff
@@ -409,3 +409,20 @@ private[chisel3] class UnsafeEnum(override val width: Width) extends EnumType(Un
   override def cloneType: this.type = new UnsafeEnum(width).asInstanceOf[this.type]
 }
 private object UnsafeEnum extends EnumFactory
+
+object suppressEnumCastWarning {
```

Comment:
```suggestion
/** Suppress enum cast warnings
  *
  * Users should use [[EnumFactory.safe <EnumType>.safe]] when possible.
  *
  * This is primarily used for casting from [[UInt]] to a Bundle type that contains an Enum.
  * {{{
  * class MyBundle extends Bundle {
  *   val addr = UInt(8.W)
  *   val op = OpEnum()
  * }
  *
  * // Since this is a cast to a Bundle, cannot use OpCode.safe                                                                                    
  * val bundle = suppressEnumCastWarning {
  *   someUInt.asTypeOf(new MyBundle)
  * }
  * }}}
  */
object suppressEnumCastWarning {
```
The best place for ScalaDoc is on the object because that's the first thing you see when you open the doc page, here's a little example of using this